probe-rs/src/architecture/riscv/mod.rs

```diff
@@ -620,6 +626,23 @@ impl<'probe> CoreInterface for Riscv32<'probe> {
             "Fpu detection not yet implemented"
         )))
     }
+
+    fn id(&self) -> usize {
+        self.id
+    }
+
+    fn reset_catch_set(&mut self) -> Result<(), Error> {
```

Comment:
I think those will make RiscV explode if we reset the core.

Reply:
I've added an (untested) implementation for reset_catch_set and reset_catch_clear.